File: SortVision/public/code/bubble/csharp/bubbleSort.cs
using System;

public static class BubbleSort {
    // Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
    public static void Sort(int[] arr) {
        if (arr == null || arr.Length <= 1) return;

        for (int i = 0; i < arr.Length - 1; i++) {
            bool swapped = false;
            for (int j = 0; j < arr.Length - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    (arr[j], arr[j + 1]) = (arr[j + 1], arr[j]);
                    swapped = true;
                }
            }
            if (!swapped) break;
        }
    }

    public static void Main() {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        Sort(arr);
        Console.WriteLine(string.Join(", ", arr));
    }
}

File: SortVision/public/code/bubble/rust/bubbleSort.rs
// Bubble Sort (in-place)
// Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
pub fn bubble_sort(arr: &mut [i32]) {
    let n = arr.len();
    for i in 0..n.saturating_sub(1) {
        let mut swapped = false;
        for j in 0..n - i - 1 {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
                swapped = true;
            }
        }
        if !swapped {
            break;
        }
    }
}

fn main() {
    let mut arr = [64, 34, 25, 12, 22, 11, 90];
    bubble_sort(&mut arr);
    println!("{:?}", arr);
}
